storybookjs/react-native
# Storylist UI usability improvements

Issue:

The storylist UI can feel cluttered, some contributing factors from my point of view are:

- Confusing and dominated visual hierarchy: sections have large headers but the section itself is not interactive and is only secondary information, large icons but the icons are only a visual hint
- "Wall of text" effect: because the sections run into one another, it creates the visual effect of a sentence without fullstops
- Easy to lose context: when focusing away from the story list (by navigating, for example), coming back takes an uncomfortable amount of time to regain mental context

## What I did

**FYI: This PR builds on https://github.com/storybookjs/react-native/pull/427**

- Resized the icons and section headers to dominate the visual hierarchy less
- Adjust story indentation so that they align with the section, and are easier to scan
- Highlight the entire section that contains the selected story, to help reestablish workflow context
- Separate sections with whitespace to create implicit groupings
- Adjust the search input to look like a search input

Extras:
- Disabled auto-correct / auto-complete on the search input, this generally just causes chaos because your search term (which is often only part of a word) gets auto-corrected to something else
- Made section headers pressable, which just selects the first story

| Before | After |
|--------|-------|
| ![image](https://user-images.githubusercontent.com/112166/219651607-04d3fa9c-55bd-4b3b-8f0b-1032f37d6c0e.png) | ![image](https://user-images.githubusercontent.com/112166/219651717-2d6172d1-b851-4de9-aa1f-9ac2c6842039.png) |

## How to test

Please explain how to test your changes and consider the following questions

- Does this need a new example in examples/native? **No.**
- Does this need an update to the documentation? **I don't think so.**

If your answer is yes to any of these, please make sure to include it in your PR.
